compiler/hash-typecheck/src/new/passes/symbol_resolution.rs

```diff
@@ -48,6 +51,66 @@ impl<'tc> SymbolResolutionPass<'tc> {
     }
 }
 
+impl<'tc> SymbolResolutionPass<'tc> {
+    /// Run a function for each stack member in the given pattern.
+    ///
+    /// The stack members are found in the `AstInfo` store, specifically the
+    /// `stack_members` map. They are looked up using the IDs of the pattern
+    /// binds, as added by the `add_stack_members_in_pat_to_buf` method of the
+    /// `ScopeDiscoveryPass`.
+    fn for_each_stack_member_of_pat(
+        &self,
+        node: ast::AstNodeRef<ast::Pat>,
+        f: impl Fn(StackMemberId) + Copy,
+    ) {
+        match node.body() {
+            ast::Pat::Binding(_) => {
+                if let Some(member_id) = self.ast_info().stack_members().get_data_by_node(node.id())
+                {
+                    f(member_id);
+                }
+            }
+            ast::Pat::Tuple(tuple_pat) => {
+                for entry in tuple_pat.fields.ast_ref_iter() {
+                    self.for_each_stack_member_of_pat(entry.pat.ast_ref(), f);
+                }
+            }
+            ast::Pat::Constructor(constructor_pat) => {
+                for field in constructor_pat.fields.ast_ref_iter() {
+                    self.for_each_stack_member_of_pat(field.pat.ast_ref(), f);
+                }
+            }
+            ast::Pat::List(list_pat) => {
+                for pat in list_pat.fields.ast_ref_iter() {
+                    self.for_each_stack_member_of_pat(pat, f);
+                }
+            }
+            ast::Pat::Or(or_pat) => {
+                if let Some(pat) = or_pat.variants.get(0) {
+                    self.for_each_stack_member_of_pat(pat.ast_ref(), f)
+                }
+            }
+            ast::Pat::Spread(spread_pat) => {
+                if let Some(name) = spread_pat.name.as_ref() {
+                    if let Some(member_id) =
+                        self.ast_info().stack_members().get_data_by_node(name.ast_ref().id())
+                    {
+                        f(member_id);
+                    }
+                }
+            }
```

Comment:
I suggest we wait until I put up my changes for the spread pattern stuff before merging this

Reply:
I don't think that will change anything?

Reply:
Yes it will, spread patterns don't exist as an AST node anymore, they are present on tuples, lists and constructors only.

Reply:
Sure, i see. However, this will have to be changed in `add_stack_members_in_pat_to_buf` as well, so it might be easier to just merge this first and then modify once your PR is up? I don't mind either way but otherwise i'd have to branch off this PR for further work

Reply:
I've already made changes in `add_stack_members_in_pat_buf` in my PR to deal with this case. It shouldn't be too long until this is in

Reply:
Okay, as long as you're about to put it up then that's fine